3rdparty/fastcv/src/fastcv_hal_imgproc.cpp

```diff
@@ -363,37 +417,46 @@ int fastcv_hal_boxFilter(
         CV_HAL_RETURN_NOT_IMPLEMENTED("ROI not supported");
     }
 
+    if(src_depth == CV_32F && normalize != 1)
+        CV_HAL_RETURN_NOT_IMPLEMENTED("normalized kernel supported for float types");
+
+    if(src_depth == CV_8U && (ksize_width != 3 && ksize_width != 5))
+        CV_HAL_RETURN_NOT_IMPLEMENTED("kernel size not supported");
+
     INITIALIZATION_CHECK;
 
-    fcvBorderType bdr;
-    uint8_t bdrVal = 0;
-    switch(border_type)
-    {
-        case cv::BORDER_REPLICATE:
-            bdr = FASTCV_BORDER_REPLICATE;
-            break;
-        case cv::BORDER_REFLECT:
-            bdr = FASTCV_BORDER_REFLECT;
-            break;
-        case cv::BORDER_REFLECT101:    // cv::BORDER_REFLECT_101, BORDER_DEFAULT
-            bdr = FASTCV_BORDER_REFLECT_V2;
-            break;
-        default:
-            CV_HAL_RETURN_NOT_IMPLEMENTED("border type not supported");
-    }
+    cv::Mat dst_temp;
+    bool inPlace = src_data == dst_data ? true : false ;
 
-    fcvStatus status = FASTCV_SUCCESS;
-    if(ksize_width == 3)
+    int nThreads = cv::getNumThreads();
+
+    cv::Mat src = cv::Mat(height, width, src_depth, (void*)src_data, src_step);
+
+    if(inPlace)
+        dst_temp = cv::Mat(height, width, src_depth);
+    else
+        dst_temp = cv::Mat(height, width, src_depth, (void*)dst_data, dst_step);
+
+    int nStripes, stripeHeight = nThreads * 10;
```

Comment:
`stripeHeight = nThreads * 10;` sounds strange. More threads -> larger piece for each thread.

Reply:
threads are 8. It's not true:
- Android build uses 2 threads by default. It's done to prevent overheating, but may be changed in future.
- Linux builds use all available cores.
- parallel_for_ serializes nested parallel_for_ calls. So you can easily get 1 here.

Reply:
We usually set granularity to some reasonable size for single thread. OpenCV uses dynamic scheduling, so all other steps are done automatically.